bait-files/src/main/webapp/submission.html

```diff
@@ -25,15 +26,15 @@ <h2>What do you know about this person?</h2>
         <ul>
             <li id='name-input'>
                 <label for="name">Full Name:</label><br>
-                <input type="text" id="name-input" name="name-input">
+                <input type="text" id="name-input" name="name-input" onkeydown="return /[a-z, ]/i.test(event.key)">
             </li>
             <li id='username-input'>
                 <label for="username">Username:</label><br>
-                <input type="text" id="username-input" name="username-input">
+                <input type="text" id="username-input" name="username-input" placeholder="@" onkeypress="checkSpace(event)">
```

Comment:
Don't you need to "return" the result of checkSpace instead of just calling it?
So: onkeypress="return checkSpace(event)"

Also, I would name the function something that tells us a little more about what we should expect out of the return value.  It seems like this method returns true if the typed character is a valid character for a user name and false otherwise.. is that right?  If so, I'd call the function something like keyIsValidForUsername.

Reply:
yep that's right - added return and changed function name!